providers/flagd/src/main/java/dev/openfeature/contrib/providers/flagd/resolver/common/ChannelBuilder.java

```diff
@@ -15,12 +16,64 @@
 import java.io.File;
 import java.net.URI;
 import java.net.URISyntaxException;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
 import java.util.concurrent.TimeUnit;
 import javax.net.ssl.SSLException;
 
 /** gRPC channel builder helper. */
 public class ChannelBuilder {
 
+    static final int MAX_RETRY_ATTEMPTS = 3;
+
+    /**
+     * Controls retry (not-reconnection) policy for failed RPCs.
+     */
+    @SuppressWarnings({"unchecked", "rawtypes"})
+    static final Map<String, ?> SERVICE_CONFIG_WITH_RETRY = new HashMap() {
```

Comment:
Let me know your thoughts on this policy. I will add it to the flagd spec once merged, since this is available in most gRPC implementations.

Note that this is used by both RPC and in-process impls, and it impacts both streams and unary RPCs (I tested this).

see: https://grpc.io/docs/guides/retry/